week-2/Homework/extra/2-array-methods.js

```diff
@@ -5,31 +5,72 @@ Following are set of small functions that you need to create using loops and arr
 //1)Find the Smallest Number in an Array
 //Create a function that takes an array of numbers and return the smallest number in it
 
-function findSmallestNum(arr) {}
+function findSmallestNum(arr) {
+  for (let i = 0; i < arr.length; i++) {
+    arr.sort((a, b) => b - a);
+  }
+  console.log(arr.pop());
+  return arr.pop();
+}
 
 findSmallestNum([354, 15, 114, 2]); //Expected: 2
 
 findSmallestNum([-76, 1, -79, 1, 0]); //Expected: -79
 
 //2)Checking Even Numbers
 //Create a function that takes in an array and returns true if all its values are even, and false otherwise.
-function checkAllEven(arr) {}
+function checkAllEven(arr) {
+  let count = 0;
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i] % 2 === 0) {
+      count++;
 
+      if (count === arr.length) {
+        console.log(true);
+        return true;
+      }
+    }
+  }
+  console.log(false);
+  return false;

```

Comment:
Same here :) you work is excellent and does work but how about giving a shot to array methods
try searching for (array.every) method